services/drone/drone-build.service.js

```diff
@@ -0,0 +1,131 @@
+'use strict'
+
+const Joi = require('joi')
+const serverSecrets = require('../../lib/server-secrets')
+const { isBuildStatus, renderBuildStatusBadge } = require('../build-status')
+const { optionalUrl } = require('../validators')
+const { BaseJsonService } = require('..')
+
+const DroneBuildSchema = Joi.object({
+  status: Joi.alternatives()
+    .try(isBuildStatus, Joi.equal('none'))
+    .required(),
+}).required()
+
+const queryParamSchema = Joi.object({
+  server: optionalUrl,
+}).required()
+
+module.exports = class DroneBuild extends BaseJsonService {
+  static get category() {
+    return 'build'
+  }
+
+  static get route() {
+    return {
+      queryParamSchema,
+      base: 'drone/build',
+      pattern: ':user/:repo/:branch*',
+    }
+  }
+
+  static get staticPreview() {
+    return {
+      message: 'passing',
+      color: 'brightgreen',
+    }
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'build',
+    }
+  }
+
+  static render({ status }) {
+    return renderBuildStatusBadge({ status })
+  }
+
+  async handle({ user, repo, branch }, { server }) {
+    const options = {
+      qs: {
+        ref: branch ? `refs/heads/${branch}` : undefined,
+      },
+    }
+    if (serverSecrets.drone_token) {
+      options.headers = {
+        Authorization: `Bearer ${serverSecrets.drone_token}`,
+      }
+    }
+    if (!server) {
+      server = 'https://cloud.drone.io'
+    }
+    const json = await this._requestJson({
+      options,
+      schema: DroneBuildSchema,
+      url: `${server}/api/repos/${user}/${repo}/builds/latest`,
+    })
```

Comment:
It looks like the API will return a 401 status code if a user specifies an unknown/invalid repo or if they specify a private repo (at least based on some quick spot testing). It may be useful to include a custom error message to handle those scenarios:

```suggestion
      errorMessages: {
        401: 'repo not found or not authorized',
      },
    })
```

Otherwise I believe folks will see a generic `invalid` badge, even if they just had a typo in their repo name
![](https://img.shields.io/badge/build-unknown-lightgrey.svg)

Reply:
Sounds great!